### PR TITLE
Fix protobuf CVE

### LINKED ISSUE #41: 🔐 CVE-2024-7254: com.google.protobuf:protobuf-java:jar:3.19.6:provided
## Summary
Any project that parses untrusted Protocol Buffers dataÂ containing an arbitrary number of nested groups / series of SGROUPÂ tags can corrupted by exceeding the stack limit i.e. StackOverflow. Parsing nested groups as unknown fields with DiscardUnknownFieldsParser or Java Protobuf Lite parser, or against Protobuf map fields, creates unbounded recursions that can be abused by an attacker.

CVE: CVE-2024-7254
CWE: CWE-20

## References
- https://ossindex.sonatype.org/vulnerability/CVE-2024-7254?component-type=maven&component-name=com.google.protobuf%2Fprotobuf-java&utm_source=ossindex-client&utm_medium=integration&utm_content=1.8.1
- http://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2024-7254
- https://github.com/advisories/GHSA-735f-pc8j-v9w8

### PR DESCRIPTION
**Pull request by Shmuma**
Closes #41

### CODE REVIEW

File: doc/changes/changes_2.0.8.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Spark Connector Common Java 2.0.8, released 2024-09-24
```
Comment (redcatbear) [inline doc/changes/changes_2.0.8.md:1]: Did you intentionally set tomorrow as release date to have some buffer? :-)

File: doc/changes/changes_2.0.8.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+Code name: Fixed vulnerability CVE-2024-7254 in com.google.protobuf:protobuf-java:jar:3.19.6:provided
+
+## Summary
+This release fixes vulnerability CVE-2024-7254 in com.google.protobuf:protobuf-java:jar:3.19.6:provided
```
Comment (redcatbear) [inline doc/changes/changes_2.0.8.md:6]:
```suggestion
This release fixes vulnerability CVE-2024-7254 in `com.google.protobuf:protobuf-java:jar:3.19.6:provided` which could lead to unbounded recursion.
```

File: pom.xml [hunk 3]
```diff
@@ -123,6 +123,12 @@
             <artifactId>janino</artifactId>
             <version>3.1.12</version>
         </dependency>
+        <dependency>
+            <!-- Upgrade transitive dependency of org.apache.spark:spark-network-common_2.13 to fix CVE-2024-7254 -->
+            <groupId>com.google.protobuf</groupId>
+            <artifactId>protobuf-java</artifactId>
+            <version>3.25.5</version>
+        </dependency>
```
Comment (redcatbear) [inline pom.xml:131]:
```suggestion
            <version>3.25.5</version>
            <scope>provided</scope>
        </dependency>
```

You might want to check if other dependencies also have the wrong scope.

File: doc/changes/changes_2.0.8.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+## Features
```
Comment (redcatbear) [inline doc/changes/changes_2.0.8.md:8]:
```suggestion
## Security
```

**Review by redcatbear — CHANGES_REQUESTED** [review]

File: doc/changes/changes_2.0.8.md (hunk 1 above)
Comment (Shmuma) [reply doc/changes/changes_2.0.8.md:1]: Yes, sometimes PK goes wild and require lots of fixing/dancing around. So, having extra day for the release normally a good idea

**Review by redcatbear — APPROVED** [review]

**Pull request merged**